### PR TITLE
Spider crawl and death sound

### PR DESCRIPTION
**Pull request by kattalainen**
Spider also has a random "noise" it makes sometimes. 
There is also a new menu song. 
Audio files to be put in "IgnoredAudio" will be put up on the drive in Kodrelaterat/Human models.../Sounds/Encrypted sounds/Spider sounds

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Components/EnemyComponent.cpp [hunk 1]
```diff
@@ -30,11 +32,21 @@ void component::EnemyComponent::Update(double dt)
...
+    if (strcmp(m_pParent->GetName().substr(0, 11).c_str(), "enemySpider") == 0)
+    {
+        m_pParent->GetComponent<component::Audio3DEmitterComponent>()->UpdateEmitter(L"SpiderCrawl");
+    }
 }
 
 void component::EnemyComponent::OnInitScene()
 {
 	m_pFactory->AddEnemyToList(m_pParent);
+    if (strcmp(m_pParent->GetName().substr(0, 11).c_str(), "enemySpider") == 0)
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/EnemyComponent.cpp]: This if statement is unnneccessary!
The sound will only play if it has been loaded to the component.

File: StortSpelProjekt/Game/src/Gamefiles/Components/EnemyComponent.cpp [hunk 2]
```diff
@@ -30,11 +32,21 @@ void component::EnemyComponent::Update(double dt)
         m_pParent->GetComponent<component::Audio3DEmitterComponent>()->UpdateEmitter(L"OnGrunt");
         m_pParent->GetComponent<component::Audio3DEmitterComponent>()->Play(L"OnGrunt");
     }
+
+    if (strcmp(m_pParent->GetName().substr(0, 11).c_str(), "enemySpider") == 0)
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/EnemyComponent.cpp]: This if statement is unneccessary!
The sound will only be updated if it has been loaded to the component

File: StortSpelProjekt/Game/src/Gamefiles/Components/EnemyComponent.cpp [hunk 3]
```diff
@@ -87,3 +99,11 @@ void component::EnemyComponent::SetRandSeed(unsigned long seed)
 {
     m_Rand.SetSeed(seed);
 }
+
+void component::EnemyComponent::death(Death* evnt)
+{
+    if (strcmp(evnt->ent->GetName().substr(0, 11).c_str(), "enemySpider") == 0)
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/EnemyComponent.cpp]: Same as before...

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 4]
```diff
@@ -199,6 +200,11 @@ Entity* EnemyFactory::Add(const std::string& entityName, EnemyComps* comps)
 		ae->AddVoice(comps->OnGruntSounds[0], L"OnGrunt");
 	}
 
+	if (comps->isSpider)
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp]: Dont like this, I would like to see an entry in EnemyComps instead to avoid if statements.

**Review by Schtekt — CHANGES_REQUESTED** [review]
Love the use of the sound!
There are a few changes I would like to see in the code though!

**Review by Schtekt — DISMISSED** [review]

**Review by LeoWikstrom — DISMISSED** [review]
Niiiiiiice 🕷️

**Review by LeoWikstrom — DISMISSED** [review]

**Review by LeoWikstrom — APPROVED** [review]

**Review by Schtekt — APPROVED** [review]

**Pull request merged**